### ISSUE TITLE
Use SDK 30 in instrumented tests with robolectric.

### ISSUE THREAD

**Issue by mcarare** [mozilla-mobile/android-components#9602]
In order to use SDK 30 in tests, we need to upgrade robolectric dependency to [4.5](https://github.com/robolectric/robolectric/releases/tag/robolectric-4.5).

This in change requires Java 9+ so this also depends on https://github.com/mozilla-mobile/android-components/issues/9579.

**Comment by pocmo** [comment]
This should be unblocked now? :)

**Comment by mcarare** [comment]
@pocmo I think we also need to upgrade robolectric. 🤖

**Comment by mcarare** [comment]
Closing in favor of https://github.com/mozilla-mobile/android-components/issues/11670.
